simpeg_drivers/joint/params.py

```diff
@@ -28,16 +36,116 @@ class BaseJointOptions(BaseInversionOptions):
     :param group_c_multiplier: Multiplier for the data misfit function for Group C.
     """
 
-    physical_property: str = ""
-    data_object: None = None
+    model_config = ConfigDict(frozen=False)
+
+    icon: str | None = None
+    documentation: str | None = None
+    version: str = simpeg_drivers.__version__
+    run_command: str = "simpeg_drivers.driver"
+    conda_environment: str = "simpeg-drivers"
+    forward_only: bool = False
+    physical_property: list[str] = ["conductivity"]
+
     group_a: SimPEGGroup
     group_a_multiplier: float = 1.0
     group_b: SimPEGGroup
     group_b_multiplier: float = 1.0
     group_c: SimPEGGroup | None = None
     group_c_multiplier: float | None = None
 
+    mesh: Octree | DrapeModel | None
+
+    active_cells: ActiveCellsOptions
+    tile_spatial: int = 1
+    parallelized: bool = True
+    solver_type: SolverType = SolverType.Pardiso
+    save_sensitivities: bool = False
+    n_cpu: int | None = None
+    max_chunk_size: int = 128
+    out_group: SimPEGGroup | UIJsonGroup | None = None
+    generate_sweep: bool = False
+    distributed_workers: str | None = None
+    alpha_s: float | FloatData | None = 1.0
+    length_scale_x: float | FloatData = 1.0
+    length_scale_y: float | FloatData | None = 1.0
+    length_scale_z: float | FloatData = 1.0
+    s_norm: float | FloatData | None = 0.0
+    x_norm: float | FloatData = 2.0
+    y_norm: float | FloatData | None = 2.0
+    z_norm: float | FloatData = 2.0
+    gradient_type: str = "total"
+    max_irls_iterations: int = 25
+    starting_chi_factor: float = 1.0
+
+    chi_factor: float = 1.0
+    auto_scale_misfits: bool = True
+    initial_beta_ratio: float | None = 100.0
+    initial_beta: float | None = None
+    cooling_factor: float = 2.0
+
+    cooling_rate: float = 1.0
+    max_global_iterations: int = 50
+    max_line_search_iterations: int = 20
+    max_cg_iterations: int = 30
+    tol_cg: float = 1e-4
+    f_min_change: float = 1e-2
+    solver_type: SolverType = SolverType.Pardiso
+
+    sens_wts_threshold: float = 1e-3
+    every_iteration_bool: bool = True
+
+    store_sensitivities: str = "ram"
+
+    beta_tol: float = 0.5
+    percentile: float = 95.0
+    epsilon_cooling_factor: float = 1.2
+
     @property
     def groups(self) -> list[SimPEGGroup]:
         """List all active groups."""
         return [k for k in [self.group_a, self.group_b, self.group_c] if k is not None]
+
+    @field_validator("n_cpu", mode="before")
+    @classmethod
+    def maximize_cpu_if_none(cls, value):
+        if value is None:
+            value = int(multiprocessing.cpu_count())
+        return value
```

Comment:
Why this if n_cpu is typed as `int | None`

Reply:
cleaned out along with parallelized